controllers/trainingjob/trainingjob_controller_test.go

```diff
@@ -617,6 +617,296 @@ var _ = Describe("Reconciling a TrainingJob that exists", func() {
 				})
 			})
 		})
+
+		Context("TrainingJob is 'Failed' and has debug status 'InProgress'", func() {
+			var failureReason string
+			var debugJobStatus sagemaker.RuleEvaluationStatus
+
+			BeforeEach(func() {
+				expectedStatus = sagemaker.TrainingJobStatusFailed
+				failureReason = "Failure within the training job"
+				debugJobStatus = sagemaker.RuleEvaluationStatusInProgress
+
+				// Add the failure reason to the describe output
+				describeOutput := CreateDescribeOutputWithDebugJobStatus(expectedStatus, expectedSecondaryStatus, debugJobStatus)
+				describeOutput.FailureReason = ToStringPtr(failureReason)
+
+				mockSageMakerClientBuilder.
+					AddDescribeTrainingJobResponse(describeOutput)
+			})
+
+			When("!HasDeletionTimestamp", func() {
+				It("Requeues after interval", func() {
+					ExpectRequeueAfterInterval(reconcileResult, reconcileError, pollDuration)
+				})
+
+				It("Updates status", func() {
+					ExpectJobsStatusToBe(trainingJob, string(expectedStatus), string(expectedSecondaryStatus), string(debugJobStatus))
```

Comment:
Should we create a second helper function `ExpectDebugJobStatusToBe` so that we don't need to bother unit testing the debug statuses in every other test?

Reply:
I thought that way but then code duplications remains same.

Reply:
I've renamed, hope that will be clear.